### PR TITLE
Feature/actions docker

### PR DESCRIPTION
**Pull request by BatteryRage**
Create docker image and push to ghcr.io

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -1,45 +1,43 @@
...
-      with:
-        node-version: 18
-        cache: 'npm'
-    - run: npm ci
-    - run: npm run start
+      - name: Check out the repo
+        uses: actions/checkout@v3
+
+      - name: Log in to GHCR
+        uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
```
Comment (sethfaulkner) [inline .github/workflows/deploy.yml]: No need to put the hash. Just do

`docker/login-action@v2`

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -1,45 +1,43 @@
...
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GHCR_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
```
Comment (sethfaulkner) [inline .github/workflows/deploy.yml]: `v4`

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -1,45 +1,43 @@
...
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+
+      - name: Build and push Docker image
+        uses: docker/build-push-action@ad44023a93711e3deb337508980b4b5e9bcdc5dc
```
Comment (sethfaulkner) [inline .github/workflows/deploy.yml]: `v3`

**Comment by BatteryRage** [comment]
@sethfaulkner Updated and removed hashes -- tested build and still functioning

**Pull request merged**